tests/Tests/500__jupyterhub/culler.robot

```diff
@@ -0,0 +1,169 @@
+    # grep from log -> td2 = timedelta(hours=HH, minutes=MM, seconds=SS); td2>td 
+
+Verify Culler Does Not Kill Active Server
+    [Documentation]    Verifies that the culler does not kill an active 
+    ...    server even after timeout has passed.
+    [Tags]  Sanity    Tier1
+    ...    ODS-1253
+    Spawn Minimal Image
+    Clone Git Repository And Open    https://github.com/redhat-rhods-qe/ods-ci-notebooks-main    notebooks/500__jupyterhub/Active.ipynb
+    Open With JupyterLab Menu    Run    Run All Cells
+    Open With JupyterLab Menu    File    Save Notebook
+    Close Browser
+    Sleep    ${${CUSTOM_CULLER_TIMEOUT}+60}
+    ${notebook_pod_name} =  Get User Notebook Pod Name  ${TEST_USER.USERNAME}
+    OpenShiftLibrary.Search Pods  ${notebook_pod_name}  namespace=rhods-notebooks
+    # Verify Culler Logs ?
+
+Verify Do Not Stop Idle Notebooks
+    [Documentation]    Disables the culler (default configuration) and verifies nb is not culled
+    [Tags]    Sanity    Tier1
+    ...    ODS-1230
+    Disable Culler
+    Close Browser
+    Spawn Minimal Image
+    Run Cell And Check Output  print("Hello World")  Hello World
+    Open With JupyterLab Menu    File    Save Notebook
+    Close Browser
+    Sleep    ${${CUSTOM_CULLER_TIMEOUT}+60}
+    ${notebook_pod_name} =  Get User Notebook Pod Name  ${TEST_USER.USERNAME}
+    OpenShiftLibrary.Search Pods  ${notebook_pod_name}  namespace=rhods-notebooks
+
+*** Keywords ***
+Spawn Minimal Image
+    [Documentation]    Spawn a minimal image
+    Begin Web Test
+    Launch JupyterHub Spawner From Dashboard
+    Spawn Notebook With Arguments  image=s2i-minimal-notebook  size=Default
+
+Get Culler Pod
+    [Documentation]    Finds the current culler pod and returns the name
+    ${culler_pod} =  OpenShiftLibrary.Oc Get  kind=Pod  label_selector=app=jupyterhub-idle-culler  namespace=redhat-ods-applications
+    ${length} =  Get Length  ${culler_pod}
+    # Only 1 culler pod, correct one
+    IF  ${length}==1
+        ${culler_pod_name} =  Set Variable  ${culler_pod[0]}[metadata][name]
+    ELSE
+    # There can be more than one during rollout
+        Sleep  10s
+        ${culler_pod_name} =  Get Culler Pod
+    END
+    Log  ${culler_pod}
+    Log  ${culler_pod_name}
+    [Return]  ${culler_pod_name}
+
+Get Culler Timeout
+    [Documentation]    Gets the current culler timeout
+    ${current_timeout} =  OpenShiftLibrary.Oc Get  kind=ConfigMap  name=jupyterhub-cfg  namespace=redhat-ods-applications  fields=['data.culler_timeout']
+    ${current_timeout} =  Set Variable  ${current_timeout[0]['data.culler_timeout']}
+    Log  ${current_timeout}
+    Log To Console  ${current_timeout}
+    ${CULLER_POD} =  Get Culler Pod
+    ${culler-env-timeout} =  Run  oc exec ${CULLER_POD} -n redhat-ods-applications -- printenv CULLER_TIMEOUT
+    Should Be Equal  ${current_timeout}  ${culler-env-timeout}
+    [Return]  ${current_timeout}
+
+Modify Culler Timeout
```

Comment:
This keyword could be in ODHDashboard.robot and be renamed to `Set Notebook Culler Timeout`

Reply:
Renamed the keyword